# InfectiousPleiotropies
First analysis of my PhD in the study of Genomic Evolution of Longevity

IN THIS ANALYSIS, I HAVE BEEN FOLLOWING THESE STEPS (PIPELINE)

1- Filter GWAS Catalog in order to find godd-quality associations 
  --> Pandas/Numpy arrays
  
2- Scripts done to retrieve single SNPs associated with 2+ Diseases
 
  
3- Create sqlite3 DB with the different SNP associated Pairs
   --> MySQL/SQLite3
   
4- Get all possible pleiotropies and classify them
   --> Python3/plink
   
   
5- Do Chi-Square test to observe any excess of AP pleiotropies 
   --> R software
 
